docs/proposals/PROPOSAL_D.md

```diff
@@ -0,0 +1,480 @@
+  - Defines namespace for their community metadata
+  - Maintains a schema "compspec.json" in a web-accessible location for their community metadata
+  - Optionally maintains plugins or associated tooling
+- A **Compatibility Schema** is a set of namespaced metadata that describes a compatibiilty interest group's space of expertise.
+  - Maintained by a compatibility interest group
+  - Metadata is described in a graph format, [Json Graph Format](https://github.com/jsongraph/json-graph-specification), version 2.0, with edges and nodes required.
+  - The graph id is the namespace or label for their community metadata (e.g,. `supercontainers.org` or `archspec.io`)
+  - Every node must be represented in the graph (defined in edges), and there should be common validation for this.
+  - Each nodes is a specific metadata attribute (e.g., `cpu` and `cpu.family`)
+  - Each edge is a relationship between two nodes (e.g., `cpu` contains -> `cpu.family`.
+- A **Compatibility Specification** is compatibility metadata that describes a container image or application
+  - Includes one or more compatibility schemas
+  - Also is validated by way of links to the compspec.json for each
+
+Importantly, there is no complex graph structure in artifact itself, but rather in the single, top level schema that defines the space of metadata attributes and relationships. The tools can use the graph (nodes and edges) as needed for their use case. This means they _are not required_ to use them and can simply view the compatibility spec (what we might refer to as an artifact) as a listing of metadata attributes. This gives power and freedom to tool developers to create tools as they see fit. Several designs are afforded:
+
+1. **Simple checking of key/value pairs**: A community might have an algorithm that is as simple as their tool checking for particular keys, and that values match for a host of interest.
+2. **Graph-based logic**: A community might warrant using advanced graph parsing to do a similar assessment. An example comes from [Archspec](https://github.com/archspec), where a single architecture (from a flat artifact or the platform, depending on what the tool chooses) can [be placed in a graph](https://github.com/archspec/archspec-json/blob/master/cpu/microarchitectures.json) and used for advanced assessment of compatibility.
+
+## Modifications
+
+The compatibility schema that defines the structure of the artifacts and relationships between attributes would need to live in a single, known location, a web accessible URL that would be maintained by the compatibility interest group, and would be versioned. The artifacts would reference the version they are using. The compatibility artifact JSON blob would live alongside images in a registry.
+
+## Compatibility Schema
+
+The compatibility schema is a single source of truth (with versions) that lives in a web accessible location and defines, for a compatibility interest group:
```

Comment:
Assuming we have our own repository with the tool. Would you allow to keep the schema out of our repository? For example if we start importing NFD and we build a schema for it, where do you think it should be placed? On the external or our repo?

Reply:
Yes this is what we are already doing. A community like supercontainers or archspec owns the schema (nodes and relationships). Here is an example:

```
{
  "version": "0.0.0",
  "kind": "CompatibilitySpec",
  "metadata": {
    "name": "lammps-prototype",
    "schemas": {
      "io.archspec": "https://raw.githubusercontent.com/supercontainers/compspec/main/archspec/compspec.json",
      "org.supercontainers": "https://raw.githubusercontent.com/supercontainers/compspec/main/supercontainers/compspec.json"
    }
  },
  "compatibilities": [
    {
      "name": "org.supercontainers",
      "version": "0.0.0",
      "attributes": {
        "hardware.gpu.available": "yes",
        "mpi.implementation": "mpich",
        "mpi.version": "4.1.1",
        "os.name": "Ubuntu 22.04.3 LTS",
        "os.release": "22.04.3",
        "os.vendor": "ubuntu",
        "os.version": "22.04"
      }
    },
    {
      "name": "io.archspec",
      "version": "0.0.0",
      "attributes": {
        "cpu.model": "13th Gen Intel(R) Core(TM) i5-1335U",
        "cpu.target": "amd64",
        "cpu.vendor": "GenuineIntel"
      }
    }
  ]
}
```
And the schemas are defined here https://github.com/supercontainers/compspec despite the tool being elsewhere (compspec-go). Note that archspec will eventually be moved into the archspec org, it's just easier to prototype in one place for now.